### ISSUE TITLE
Problem with updating parameters

### ISSUE THREAD

**Issue by antoniaih** [pharmpy/modelbuilder#133]
Sometimes parameters get updated. For example when adding a dataset. Or when adding a transit compartment.

This is because when the model structure is changed new parameters might be added or existing parameters might be removed. Therefore `model_state.parameters` is updated to include new parameters and unused ones are removed. In other words the intersection between the model_state parameters and the model parameters (the model that is being built from the model state).
Parameters that exist in both are kept as they are. So a parameter whose initial value is different in the model than in `model_state.parameters` is changed so that it matches the one in `model_state.parameters`.

The problem is to know which set is the correct one. When we always update from the model it is not possible to change parameters in the parameter tab. Ideally we would want to know if the model features have been changed and in that case set `model_state.parameters = model.parameters`. But we do not know if the model features have been changed since there is no memory of a previous state. We would have to evaluate this outside of the model state.

**Comment by rikardn** [comment]
Do you mean that a modeling function might change a parameter value that overrides what the user has set in the UI?

**Comment by antoniaih** [comment]
Yes. But not only the user input but also the previous parameter state in general. I do not override existing parameters.

**Comment by rikardn** [comment]
Ok. What do we want to happen? One strategy is to save all parameters manually input by the user and let these override whatever is created automatically. The ones that were never changed manually we can allow being changed by the model transformations.

**Comment by antoniaih** [comment]
But they cannot be saved inside the model state because then we do not know if they have been recently changed or not. A parameter changed by the user could still be updated by the modeling functions.

**Comment by rikardn** [comment]
So we keep all changed parameters by the user and these always takes precedence (unless removed) over automatically changed parameters. So there will be a `dict` with user changed parameters and a `set_initial_estimates` will be called to keep these values. If a parameter gets removed we could consider keeping it here in case it reappears.

**Comment by antoniaih** [comment]
The current solution for this is that `model_state.parameters` gets reset whenever an attribute except parameters is updated. 
This means that when changing anything in the model after having changed parameters will overwrite the parameters. Maybe there will be a better solution for this in the future.
